### PR TITLE
Add parent_test field in SubTestMessage 

### PR DESCRIPTION
**Pull request by pupacha**
To identify which lisa test the SubTestMessage is coming from, store LISA testname that is generating the SubTestMessage into 'parent_test' field

### CODE REVIEW

**Comment by pupacha** [comment]
@anirudhrb Please review

**Comment by anirudhrb** [comment]
Is it useful to do this in a common place so that all substests automatically have this field populated?

**Comment by pupacha** [comment]
> Is it useful to do this in a common place so that all substests automatically have this field populated?

How about following change in `create_test_result_message` 
https://github.com/microsoft/lisa/blob/bb1cacf165980bdc698af4071cd6cddf9b08aa71/lisa/messages.py ? @squirrelsc @LiliDeng @anirudhrb 

    if message_type == SubTestMessage and "parent_test" not in message.information:
        message.information["parent_test"] = test_result.runtime_data.name

File: lisa/messages.py [hunk 1]
```diff
@@ -316,4 +316,6 @@ def create_test_result_message(
     message.elapsed = test_result.get_elapsed()
     if other_fields:
         dict_to_fields(other_fields, message)
+    if message_type == SubTestMessage and "parent_test" not in message.information:
```
Comment (squirrelsc) [inline lisa/messages.py]: It should be a field on SubTestMessage.

Comment (pupacha) [reply lisa/messages.py]: Made it as a field in SubTestMessage

**Review by anirudhrb — APPROVED** [review]

File: lisa/messages.py [hunk 2]
```diff
@@ -314,6 +315,10 @@ def create_test_result_message(
     message.status = test_status
     message.message = test_message
     message.elapsed = test_result.get_elapsed()
+    if message_type == SubTestMessage:
+        if not other_fields:
+            other_fields = {}
+        other_fields.update({"parent_test": test_result.runtime_data.name})
```
Comment (LiliDeng) [inline lisa/messages.py:321]: it should be filled in 
other_fields["information"]["parent_test"]

this is current data run from your changed code
```
2023-04-12 14:25:32.872[5320][DEBUG] lisa.notifier[Console] received message [SubTestResult]: SubTestMessage(type='SubTestResult', time=datetime.datetime(2023, 4, 12, 14, 25, 32, 794499), elapsed=2972.2109429, id_='lisa_0_0', name='binfmt_misc02', status=<TestStatus.PASSED: 5>, message='', stacktrace=None, information={'version': '20200930', 'exit_value': 0}, hardware_platform='x86_64', parent_test='ltp_lite')
```

Comment (pupacha) [reply lisa/messages.py:321]: I initially used 'information' field to store 'parent_test' but Chi suggested to add a field in SubTestMessage itself instead. And I agree with that idea as well.

Comment (squirrelsc) [reply lisa/messages.py:321]: It's a common requirement to know the parent test name, so I asked to add it to subtest message level.

**Review by LiliDeng — APPROVED** [review]

**Pull request merged**